### PR TITLE
[#91] Add mutation to createResponse

### LINKED ISSUE #91: [Backend] As a user, I can submit survey answers
## Why
We need to prepare the backend integration to submit the answer.

## Acceptance Criteria
Create graphql mutation on the survey to update the answer

### PR DESCRIPTION
**Pull request by sleepylee**
Closes #91 

## What happened 👀

Add backend integration to submit answer
 
## Insight 📝

- It's like a POST request, in GraphQL it's a Mutation.
- Here is an example:
```
mutation createResponse($input: CreateResponseMutationInput!) {
  createResponse(input: $input){
    clientMutationId
	}
}

// Then adding the variable:
{
  "input": {
    "response": {
      "surveyId": "d5de6a8f8f5f1cfe51bc",
      "questions": [
        {
          "id": "940d229e4cd87cd1e202",
          "answers": [
            {
            "id": "037574cb93d16800eecd",
            "answer": "50"
            },
            {
            "id": "037574cb93d16800eecd",
            "answer": "50"
          }
          ]
        }
      ]
    }
  }
}
```
 
## Proof Of Work 📹
Successfully call:

### CODE REVIEW

**Review by markgravity — APPROVED** [review]
LGTM 🤖

File: lib/use_cases/create_response_use_case.dart [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import 'package:survey/api/graphql/mutation/create_response_mutation_input.dart';
+import 'package:survey/repositories/survey_repository.dart';
+
+import 'base_use_case.dart';
+
+class CreateResponseUseCase extends UseCase<bool, ResponseInput> {
```
Comment (manh-t) [inline lib/use_cases/create_response_use_case.dart:6]: perhaps `CreateSurveyResponseUseCase` is better to understand the context of response in this case 😄

File: lib/api/graphql/mutation/create_response_mutation_input.dart [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'create_response_mutation_input.g.dart';
+
+@JsonSerializable()
+class CreateResponseMutationInput {
```
Comment (manh-t) [inline lib/api/graphql/mutation/create_response_mutation_input.dart:6]: same here `CreateSurveyResponseMutationInput` or shorter `CreateSurveyResponseInput` 🙏

**Comment by sleepylee** [comment]
@manh-t it wasn't in version control yet...I committed it now 🙇

File: lib/api/graphql/mutation/create_response_mutation_input.dart (hunk 2 above)
Comment (sleepylee) [reply lib/api/graphql/mutation/create_response_mutation_input.dart:6]: this follows the convention from the graphQL endpoint 🤔  I think I'd update the UseCase name but not this one 
image

**Review by manh-t — APPROVED** [review]

**Pull request merged**